libs/lz4/src/main/java/org/elasticsearch/lz4/ESLZ4Compressor.java

```diff
@@ -0,0 +1,256 @@
+                    // Modified to use explicit == false
+                } while(LZ4SafeUtils.readIntEquals(src, ref, sOff) == false);
+
+                excess = LZ4SafeUtils.commonBytesBackward(src, ref, sOff, srcOff, anchor);
+                sOff -= excess;
+                ref -= excess;
+                int runLen = sOff - anchor;
+                int tokenOff = dOff++;
+                if (dOff + runLen + 8 + (runLen >>> 8) > destEnd) {
+                    throw new LZ4Exception("maxDestLen is too small");
+                }
+
+                if (runLen >= 15) {
+                    SafeUtils.writeByte(dest, tokenOff, 240);
+                    dOff = LZ4SafeUtils.writeLen(runLen - 15, dest, dOff);
+                } else {
+                    SafeUtils.writeByte(dest, tokenOff, runLen << 4);
+                }
+
+                LZ4SafeUtils.wildArraycopy(src, anchor, dest, dOff, runLen);
+                dOff += runLen;
+
+                while(true) {
+                    SafeUtils.writeShortLE(dest, dOff, (short)(sOff - ref));
+                    dOff += 2;
+                    sOff += 4;
+                    ref += 4;
+                    int matchLen = LZ4SafeUtils.commonBytes(src, ref, sOff, srcLimit);
+                    if (dOff + 6 + (matchLen >>> 8) > destEnd) {
+                        throw new LZ4Exception("maxDestLen is too small");
+                    }
+
+                    sOff += matchLen;
+                    if (matchLen >= 15) {
+                        SafeUtils.writeByte(dest, tokenOff, SafeUtils.readByte(dest, tokenOff) | 15);
+                        dOff = LZ4SafeUtils.writeLen(matchLen - 15, dest, dOff);
+                    } else {
+                        SafeUtils.writeByte(dest, tokenOff, SafeUtils.readByte(dest, tokenOff) | matchLen);
+                    }
+
+                    if (sOff > mflimit) {
+                        anchor = sOff;
+                        break label53;
+                    }
+
+                    SafeUtils.writeShort(hashTable, LZ4Utils.hash64k(SafeUtils.readInt(src, sOff - 2)), sOff - 2 - srcOff);
+                    int h = LZ4Utils.hash64k(SafeUtils.readInt(src, sOff));
+                    ref = srcOff + SafeUtils.readShort(hashTable, h);
+                    SafeUtils.writeShort(hashTable, h, sOff - srcOff);
+                    // Modified to use explicit == false
+                    if (LZ4SafeUtils.readIntEquals(src, sOff, ref) == false) {
+                        anchor = sOff++;
+                        break;
+                    }
+
+                    tokenOff = dOff++;
+                    SafeUtils.writeByte(dest, tokenOff, 0);
+                }
+            }
+        }
+
+        dOff = LZ4SafeUtils.lastLiterals(src, anchor, srcEnd - anchor, dest, dOff, destEnd);
+        return dOff - destOff;
+    }
+
+    public int compress(byte[] src, int srcOff, int srcLen, byte[] dest, int destOff, int maxDestLen) {
```

Comment:
I believe `@Override` is missed here since we override the `compress` method from `LZ4Compressor`